src/app/Providers/components/AddEditProviderModal/AddEditProviderModal.tsx

```diff
@@ -195,6 +199,39 @@ export const AddEditProviderModal: React.FunctionComponent<IAddEditProviderModal
     ? createProviderMutation
     : patchProviderMutation;
 
+  const [hostname, setHostname] = React.useState(fields?.hostname ? fields.hostname.value : '');
+
+  function assertIsCertificate(
+    certificate: ITLSCertificate
+  ): asserts certificate is ITLSCertificate {
+    if (!('fingerprint' in certificate)) {
+      throw new Error('Not certificate');
+    }
+  }
+
+  const getCertificate = async (hostname: string) => {
+    const response = await fetch(`get-certificate?url=${hostname}`);
+    if (!response.ok) {
+      throw new Error('Problem fetching data');
+    }
+    const certificate = await response.json();
+    assertIsCertificate(certificate);
+
+    return certificate;
+  };
+
+  const [isCertificateQueryEnabled, setCertificateQueryFlag] = React.useState(false);
+
+  const [isCertificateValid, setCertificateValid] = React.useState(false);
+
+  const { status, error, data } = useQuery<ITLSCertificate, Error>(
```

Comment:
As you suggested on Slack, I do think we should probably move this stuff to a `useCertificateQuery` hook so we're not cluttering this component. `assertIsCertificate` and `getCertificate` could be moved into there. I think we should also have the query result as a single variable instead of destructuring it, so we can use `certificateQuery.status` instead of `status` so the usage is more readable below.